src/app/createlog/page.tsx

```diff
@@ -0,0 +1,232 @@
+  const [selectedExercises, setSelectedExercises] = useState([]);
+
+  const exercisesArray = Object.values(ExercisesDictionary);
+
+  useEffect(() => {
+    if (searchInput) {
+      const results = exercisesArray.filter((exercise) =>
+        exercise.label.toLowerCase().includes(searchInput.toLowerCase())
+      );
+      setSearchResults(results);
+    } else {
+      setSearchResults([]);
+    }
+  }, [searchInput]);
+
+  const handleSelectExercise = (exercise) => {
+    setSelectedExercises((prev) => [
+      ...prev,
+      { ...exercise, sets: [{ reps: 0, weight: 0 }] },
+    ]);
+    setSearchInput(""); // Clear search input after selection
+    setSearchResults([]); // Clear search results after selection
+  };
+
+  const handleAddSet = (index) => {
+    const newSelectedExercises = [...selectedExercises];
+    newSelectedExercises[index].sets.push({ reps: 0, weight: 0 });
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  const handleSetChange = (index, setIndex, field, value) => {
+    const newSelectedExercises = [...selectedExercises];
+    newSelectedExercises[index].sets[setIndex][field] = value;
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  const handleDeleteSet = (index, setIndex) => {
+    const newSelectedExercises = [...selectedExercises];
+    // Remove the set at setIndex from the selected exercise
+    newSelectedExercises[index].sets.splice(setIndex, 1);
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  const handleSaveLog = async () => {
+    // Create sessions array
+    const sessions = [
+      {
+        date: new Date(), // Use current date as an example
+        exercises: selectedExercises.map((exercise) => {
+          // Map selected exercises to exerciseSchema
+          return {
+            exerciseName: exercise.label,
+            sets: exercise.sets.map((set, index) => {
+              // Map sets to setSchema
+              return {
+                setNumber: index + 1, // Set number starts from 1
+                weight: set.weight,
+                unit: "lbs", // Assuming the default unit is pounds
+                reps: set.reps,
+              };
+            }),
+          };
+        }),
+      },
+    ];
+
+    const formData = {
+      userId: session?.user?._id,
+      sessions: sessions,
+    };
+
+    const jsonData = JSON.stringify(formData);
+    console.log(jsonData);
+
+    try {
+      // Send a POST request to your backend API endpoint
+      const response = await fetch("/api/logging", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: jsonData,
+      });
+
+      // Check if the request was successful
+      if (response.ok) {
+        console.log("Log saved successfully!");
+      } else {
+        console.error("Failed to save log:", response.statusText);
+      }
+    } catch (error) {
+      console.error("An error occurred while saving the log:", error);
+    }
```

Comment:
What are your thoughts on moving the fetching logic to their own client file? (similar to `app/clients/auth-client/auth-client.ts`
You could create a `LoggingClient` and create a logging function and move this logic there?

Reply:
I agree with this! I just left the code here for Alan since he'll take a closer look at the backend, so I believe he will make this update.